src/apps/backend/modules/vendor_account/internal/vendor_account_reader.py

```diff
@@ -0,0 +1,16 @@
+from modules.vendor_account.errors import VendorAccountWithSameNameAndAccountExistsError
+from modules.vendor_account.internal.store.vendor_account_repository import VendorAccountRepository
+from modules.vendor_account.types import NameAlreadyExistParams, VendorAccount
+
+
+class VendorAccountReader:
+    @staticmethod
+    def check_name_not_exist(params: NameAlreadyExistParams) -> VendorAccount:
+        vendor_account = VendorAccountRepository.collection().find_one(
+            {"account": params.account_id, "name": params.name, "active": True}
```

Comment:
Based on this query pattern, I think the index should be on `account_id` and `active`. In your repository, you are creating index on `account` which does not look right.

Reply:
I think it should be on `account` , `name` and `active`.